### DISCUSSION TITLE
[Help needed] PHP session cookies during login

### DISCUSSION THREAD

**Discussion by agentclark** [FreshRSS/FreshRSS#6201] [General]
**Describe the bug**
Getting "The web server seems to be incorrectly configured for cookies required for PHP sessions!" during login.

**Expected behavior**
Access FreshRSS

**Environment information Server:**
- OS: Debian 11
- FreshRSS version: 1.23.2
- Database version: mariadb  Ver 15.1 Distrib 10.5.23-MariaDB
- PHP version: PHP 7.4
- Installation type: git

**Environment information Client:**
- OS: Windows and Android
- Browser: Firefox, Edge, Chrome, Kiwi

**Additional context**
I was running FreshRSS smoothly for a couple of years now. Yesterday I did a long due update from 1.21.1 to 1.23.2. Since then I am unable to login on all my devices except for one. On the one device I still can login, I can only login via Firefox and no other Browser.

Before the update I hadn't touched my apache configuration. I now switched from prefork to event with no solution to the problem. FreshRss is running as a subdomain, so no reverse proxy is involved. I tried session.php as mentioned in [5973](https://github.com/FreshRSS/FreshRSS/issues/5973) and it is counting as expected.

In the apache access log I get a 403 and nothing in error log. 
FreshRSS shows `[Wed, 13 Mar 2024 11:04:12 +0000] [warning] --- Invalid session during login for user=xxx, nonce=
` in ./data/users/_/log.txt

I've whiped FreshRSS completly, except for the mariadb database (which I really would love to keep for the favorites) and freshly installed from git.

Thank you so much in advance!

edit: Added FreshRSS log

**Comment by joshuacwnewton** [comment]
I was getting this exact issue as well (no reverse proxy, exact same browser login error message, same warning in the users log.txt).

What helped me in the short term:

- Clear cookies in browser.
- If using an app with the API integration, change your API password then re-auth.

For me, the issue has been intermittent. It might have been tied to a change in port as I was setting up my docker compose, thus resulting in an outdated `base_url`? :thinking: 

I'm still looking for a more permanent fix. [This forum post](https://forums.unraid.net/topic/40740-support-linuxserverio-freshrss/?do=findComment&comment=1353148) seems to be pointing in the direction of `www/freshrss/data/config.php`. Here's mine for reference:

<details>

```php
<?php
return array (
  'environment' => 'production',
  'salt' => 'xxx',
  'base_url' => 'http://xxx',
  'auto_update_url' => 'https://update.freshrss.org',
  'language' => 'en',
  'title' => 'FreshRSS',
  'meta_description' => '',
  'logo_html' => '',
  'default_user' => 'xxx',
  'force_email_validation' => false,
  'allow_anonymous' => false,
  'allow_anonymous_refresh' => false,
  'auth_type' => 'form',
  'http_auth_auto_register' => true,
  'http_auth_auto_register_email_field' => '',
  'api_enabled' => true,
  'unsafe_autologin_enabled' => false,
  'simplepie_syslog_enabled' => true,
  'pubsubhubbub_enabled' => false,
  'allow_robots' => false,
  'allow_referrer' => false,
  'limits' => 
  array (
    'cookie_duration' => 7776000,
    'cache_duration' => 800,
    'timeout' => 20,
    'max_inactivity' => 9223372036854775807,
    'max_feeds' => 131072,
    'max_categories' => 16384,
    'max_registrations' => 1,
  ),
  'curl_options' => 
  array (
  ),
  'db' => 
  array (
    'type' => 'sqlite',
    'host' => '',
    'user' => '',
    'password' => '',
    'base' => '',
    'prefix' => '',
    'connection_uri_params' => '',
    'pdo_options' => 
    array (
    ),
  ),
  'mailer' => 'mail',
  'smtp' => 
  array (
    'hostname' => '',
    'host' => 'localhost',
    'port' => 25,
    'auth' => false,
    'auth_type' => '',
    'username' => '',
    'password' => '',
    'secure' => '',
    'from' => 'root@localhost',
  ),
  'extensions_enabled' => 
  array (
  ),
  'extensions' => 
  array (
  ),
  'disable_update' => true,
  'trusted_sources' => 
  array (
    0 => '127.0.0.0/8',
    1 => '::1/128',
  ),
);
```

</details>

If I run into the issue again, I'll try to troubleshoot further to find the root cause.

**Comment by math-GH** [comment]
just for documentation a link to another issue from the past: https://github.com/FreshRSS/FreshRSS/issues/5259

**Comment by bit15k** [comment]
I have encountered the same problem. I originally used nginx and placed freshrss in a subdirectory, it worked properly. Until I changed the domain name and modified the server name of nginx, this problem occurred. So I switched back to the nginx configuration file of the old domain name, but at this point, I was unable to log in. Always display "The web server sessions to be incorrectly configured for cookies required for PHP sessions!". I am very certain that I have configured nginx, as it was running normally before.

**Reply by Alkarex** [reply]
1. Check your `base_url` in `data/config.php`
2. Try with another browser, just to make sure there is no cache problem
3. Check your Web browser network logs in the developer tools to find out what cookies you get back and which one you send in the following request

**Reply by bit15k** [reply]
The content of the `base url` is as follows, which is consistent with the` server_name` and `location` in the nginx configuration file.  
<details>

```xml
return array (
  'environment' => 'production',
  'salt' => '24088f2fb8bd2a363dca6ddccd63f192177eecce',
  'base_url' => 'https://my-domain/freshrss',
  'auto_update_url' => 'https://update.freshrss.org',
  'language' => 'en',
  'title' => 'FreshRSS',
  'meta_description' => '',
  'logo_html' => '',
  'default_user' => 'bit15k',
  'force_email_validation' => false,
  'allow_anonymous' => false,
  'allow_anonymous_refresh' => false,
  'auth_type' => 'form',
  'http_auth_auto_register' => true,
  'http_auth_auto_register_email_field' => '',
  'api_enabled' => false,
  'unsafe_autologin_enabled' => false,
  'simplepie_syslog_enabled' => true,
  'pubsubhubbub_enabled' => true,
  'allow_robots' => false,
  'allow_referrer' => false,
  'limits' => 
```

</details>

nginx configuration file reads as follows.  

<details>

```xml
...
location /freshrss/ {
        proxy_cookie_path / "/; HTTPOnly; Secure; SameSite=Lax";
        proxy_pass http://freshrss/;
        add_header X-Frame-Options SAMEORIGIN;
        add_header X-XSS-Protection "1; mode=block";
        proxy_redirect off;
        proxy_buffering off;
        proxy_set_header Host $host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-Prefix /freshrss/;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;
        proxy_set_header X-Forwarded-Port $server_port;
        proxy_read_timeout 90; 
        # Forward the Authorization header for the Google Reader API.
        proxy_set_header Authorization $http_authorization;
        proxy_pass_header Authorization;
    }
```

</details>

I tried logging in using a different local browser and it still says "The web server sessions to be incorrectly configured for cookies required for PHP sessions".    

I used the browser developer tools to check and found that clicking on login returned a `403` forbidden

![image](https://github.com/FreshRSS/FreshRSS/assets/29975541/e2a2c75c-5cc0-471c-9b03-ba8fcb97892d)

It should be noted that my freshrss was installed via docker compose, and nginx was installed on the host.

**Reply by Inverle** [reply]
@bit15k

> 'salt' => '24088f2fb8bd2a363dca6ddccd63f192177eecce',

If you are still using this salt, please generate a new one. Exposing the salt is bad for security.

Command to generate new salt:
```bash
 php -r 'echo bin2hex(random_bytes(20)) . "\n";'
```

**Comment by luke396** [comment]
I'd like to offer everyone an additional perspective; I've also been tormented by the same issue for a long time.

In my case, I can log in and access in Chrome's Incognito mode, but I can't log in in normal mode, showing a lack of cookies, just as everyone has discussed.

The problem in my situation is not with nginx, nor with the configuration files of FreshRSS, but with the settings of the Chrome browser. When all related cookie caches are deleted, and the following default configurations are modified (you may not need to modify all of them, just a few of them; forgive me for not wanting to delve further into this, as I've already spent over ten hours on this issue), everything works fine!

<img width="479" alt="{0FB33173-652A-416B-BEF1-BD4E94D4387C}" src="https://github.com/user-attachments/assets/0c0e22f1-c0cb-49e6-b2b3-1e30cd4e4adf">

<img width="466" alt="{553D82AF-E0EB-436A-B92C-AB3B5BA3DA87}" src="https://github.com/user-attachments/assets/987c83ae-d6a3-4ce0-9da9-be615ea45067">

(You can find the corresponding website in this option under "Privacy and security", and modify it! Good luck to everyone!)
<img width="539" alt="{8E65B169-DF27-4E69-8AAB-D2A6AA4A2DCC}" src="https://github.com/user-attachments/assets/0647d16a-c195-4357-b3db-e57f633280ed">

**Comment by Marcs1987** [comment]
I've ran into this same problem. 

I've just installed FreshRSS as a docker image and using Apache as a reverse proxy. After installation everything was working fine. Then, after a couple of days, I got this "The web server seems to be incorrectly configured for cookies required for PHP sessions" message when trying to log in. Then, just as suddenly, the problem went away. And now it's back again. I get the same message even if I enter an incorrect password. I've tried restarting the docker image, the Apache server and using a browser with the cookies cleared. 

Any suggestions how to tackle this problem would be appreciated

**Reply by Alkarex** [reply]
What Docker image, and what Apache configuration?

**Reply by Marcs1987** [reply]
Regarding the docker installation I followed instruction from this page
https://github.com/FreshRSS/FreshRSS/blob/edge/Docker/README.md

Regarding Apache I got the following in my ssl.conf
ProxyPass / http://127.0.0.1:8080/
ProxyPassReverse / http://127.0.0.1:8080/
RequestHeader set X-Forwarded-Prefix "/freshrss"
RequestHeader set X-Forwarded-Proto "https"

**Reply by Marcs1987** [reply]
To complement, this is what I see in Apache ssl_access_log when I try to log in

"GET /i/?c=javascript&a=nonce&user=user HTTP/1.1" 200 111
"POST /i/?c=auth&a=login HTTP/1.1" 403 2112562

at the same time I get this in the Freshrss log

[warning] --- Invalid session during login for user=user, nonce=

at the same time I don't have any problems accessing Freshrss using Readrops on my phone

**Reply by Alkarex** [reply]
Please try https://github.com/FreshRSS/FreshRSS/discussions/6201#discussioncomment-9420008

**Reply by Marcs1987** [reply]
I added the following to my ssl.conf

```
<Proxy http://127.0.0.1:8080/>
   ProxyPassReverseCookiePath / /
   ProxyPassReverseCookieDomain localhost example.com
   Require all granted
   Options none
</Proxy>
```

After that I haven't had any more problems

**Reply by Alkarex** [reply]
@Marcs1987 What `base_url` do you have in your `data/config.php` ?
Also visible in the Web interface under systems configuration https://freshrss.example.net/i/?c=configure&a=system

**Reply by Marcs1987** [reply]
it's http://my.domain.name:8080

**Reply by Alkarex** [reply]
This should be your public address, not the internal one

**Reply by Marcs1987** [reply]
You mean there shouldn't be any port number in the address? And that I should have https instead of http?

**Reply by Alkarex** [reply]
It should be exactly the same than the URL visible in your Web browser

**Reply by Marcs1987** [reply]
Tested by modifying my base_url to https://my.domain.name (that's how I see it in my browser, didn't include anything to the right of the domain name). Then I removed the Proxy configuration I made earlier and the same error as earlier reappeared :(

**Comment by lin-b-github** [comment]
I just want to report that I experienced the "cookies" error today, too. 

I use simple shared hosting. The tech support person and I performed various troubleshooting including having me attempt to connect from different browsers, browsers after caches were cleared, VPN and no VPN. All generated a "500 server error" at my end on the webpage. 

I used Brave Browser, Vilvaldi, and Pale Moon (the latter didn't provide an error, just a blank white page). The tech support didn't find any log errors in the history.

However, the tech rep said he tried to login to my public freshRSS page and received the error being discussed in this thread, 

> "The web server seems to be incorrectly configured for cookies required for PHP session."

 He just wrote back suggesing I contact the developer team of the app.

I suspect my issue could relate to my recently upgrading from an outdated 7.x PHP version that was no longer receiving security updates, to the most recent stable version offered by my hosting provider, which I believe is 8.3.

Anyway, I'll read some of the suggestions in this thread and elsewhere for solutions.

**Reply by Alkarex** [reply]
@lin-b-github Please provide some information about your setup: how you installed FreshRSS, which version, on which platform, etc.

**Reply by lin-b-github** [reply]
I emailed a response (now cut-and-pasted below), but I don't believe it transmitted to github. 

### An update on situation:

The problem was from my recent upgrade to PHP 8.4 from an outdated version 7.x PHP.  I reverted back to 7.4 PHP temporarly and then tried to log-in to FreshRSS. It worked perfectly. I could read the feeds and access the admin panel.

### Now the bad news for me:

At this point, having read some comments here or elsewhere and the Docs, I proceded to upgrade incrementally from 1.23 to 1.26 in the FreshRSS admin panel. That worked fine. 

Unfortunately, when I clicked "return-to-feed page," I received the 500 error again, to my surprise. Since I was still logged on as admin, I was still able to navigate to any section in the admin section and make modifications, but just couldn't access the feed section (it would generate an http 500 error). 

I think what might have happened is I failed to match the PHP version with the particular version of FreshRSS that I was upgrading to in the iteration process. I had left 7.4 as the PHP version as I upgraded incrementally to 1.26.

Now that I've upgraded (but probably broke something during the series of upgrades), I no longer can log-on again. I changed PHP to 7.4, 8.1, 8.2, 8.3, 8.4 -- none worked. So, I guess I broke it at this point.

### Resolution for now:

I'm not concerned. I only had 14 feeds.

Anyway, I am planning to postpone further work on this for a few days or weeks since I really don't want to continue hosting the app on my NameCheap shared hosting plan since I don't have much control to troubleshoot and fix problems (for example, no sudo/root access). 

However, if I don't find a better place to host the app, then I'll probably just delete and re-install it on my shared hosting plan. 


----
**Now outdated email:**

- Installation of FreshRSS: Nov. 2023 via Cpanel’s Softaculous automated script provided by NameCheap shared hosting provider.

- Background: I’ve only used FreshRSS a few times since installation, most recently a couple of months ago. No problems whatsoever with logging-in or using the application until two days ago.

- SSL: Although the primary domain uses SSL, I haven’t yet set-up SSL for the sub-domain that FreshRSS uses. Just pointing this out in case anyone wants to test the login error. Not sure whether this is now causing a problem, but in the past this wasn’t an issue.

- FreshRSS version: 1.23.1

- Softaculous FreshRSS version available: 1.26.3 (Mar 2025).

1. I noticed this newer version is available to install from Softaculous, but it doesn’t provide the usual automatic click update to the existing installation. 
2. I toggled the feature to automatically update when a newer version is available, but it hasn’t performed any updates. So, an upgrade to a more recent version will probably require a manual process

-     Installation platform: FreshRSS is installed as a sub-domain to a WordPress.org blog website using WordPress.org (6.8).

-     Shared Hosting technology:

        PHP: 8.4 [I updated from an outdated, no-security-updates 7.x version in the last month or so – which might be a contributing factor to the log-in problem that I’m experiencing]
        cPanel: 126.0 (build 27)
        Apache: 2.4.63
        Database: 10.6.22-MariaDB-cll-lve-log
        Operating: linux
        Kernel: 4.18.0-513.18.1.lve.el8.x86_64

**Reply by Alkarex** [reply]
* https://github.com/FreshRSS/FreshRSS/discussions/7592

**Reply by Alkarex** [reply]
Alternatives:
* https://github.com/FreshRSS/FreshRSS/issues/7676#issuecomment-2982835995
